selenium/che-selenium-test/src/test/java/org/eclipse/che/selenium/dashboard/workspaces/details/WorkspaceDetailsOverviewTest.java

```diff
@@ -0,0 +1,238 @@
+    workspaceOverview.waitErrorBorderOfNameField();
+    workspaceOverview.waitDisabledSaveButton();
+
+    // check too short name
+    setupValidName(CHANGED_WORKSPACE_NAME);
+    setupInvalidNameAndCheckErrorMessage(TOO_SHORT_NAME, SHORT_NAME_ERROR_MESSAGE);
+
+    // check too long name
+    setupValidName(MIN_SHORT_NAME);
+    setupInvalidNameAndCheckErrorMessage(TOO_LONG_NAME, LONG_NAME_ERROR_MESSAGE);
+
+    setupValidName(MAX_LONG_NAME);
+    checkValidNames();
+
+    checkInvalidNames();
+  }
+
+  @Test(priority = 1)
+  public void shouldCheckExportAsFile() {
+    workspaceOverview.checkNameWorkspace(WORKSPACE_NAME);
+
+    // check of closing by "Esc"
+    openExportWorkspaceForm();
+    seleniumWebDriverHelper.sendKeys(ESCAPE.toString());
+    workspaceOverview.waitExportWorkspaceFormClosed();
+
+    // close by "x" icon
+    openExportWorkspaceForm();
+    workspaceOverview.clickOnCloseExportWorkspaceIcon();
+    workspaceOverview.waitExportWorkspaceFormClosed();
+
+    // close by "Close" button
+    openExportWorkspaceForm();
+    workspaceOverview.clickOnCloseExportWorkspaceButton();
+    workspaceOverview.waitExportWorkspaceFormClosed();
+
+    // check config
+    openExportWorkspaceForm();
+    workspaceOverview.waitConfiguration(EXPECTED_CONFIG);
+    workspaceOverview.clickOnToPrivateCloudButton();
+    workspaceOverview.waitToPrivateCloudTabOpened();
+    seleniumWebDriverHelper.sendKeys(ESCAPE.toString());
+    workspaceOverview.waitExportWorkspaceFormClosed();
+  }
+
+  private void selectStackAndCheckWorkspaceName(NewWorkspace.Stack stack) {
+    newWorkspace.selectStack(stack);
+    newWorkspace.waitStackSelected(stack);
+    newWorkspace.waitWorkspaceNameFieldValue(WORKSPACE_NAME);
+  }
+
+  private void setupValidName(String name) {
+    workspaceOverview.enterNameWorkspace(name);
+    workspaceOverview.waitUntilNoErrorsDisplayed();
+    workspaceOverview.waitEnabledSaveButton();
+  }
+
+  private void setupInvalidNameAndCheckErrorMessage(String name, String expectedErrorMessage) {
+    workspaceOverview.enterNameWorkspace(name);
+
+    try {
+      workspaceOverview.waitErrorBorderOfNameField();
+    } catch (TimeoutException ex) {
+      // remove try-catch block after issue has been resolved
+      fail("Known issue https://github.com/eclipse/che/issues/10659", ex);
+    }
+
+    workspaceOverview.waitExpectedNameErrorMessage(expectedErrorMessage);
+    workspaceOverview.waitDisabledSaveButton();
+  }
+
+  private void checkValidNames() {
+    VALID_NAMES.forEach(
+        name -> {
+          setupInvalidNameAndCheckErrorMessage(TOO_SHORT_NAME, SHORT_NAME_ERROR_MESSAGE);
+          setupValidName(name);
+        });
+  }
+
+  private void checkInvalidNames() {
+    NOT_VALID_NAMES.forEach(
+        name -> {
+          setupValidName(CHANGED_WORKSPACE_NAME);
+          setupInvalidNameAndCheckErrorMessage(name, SPECIAL_CHARACTERS_ERROR_MESSAGE);
+        });
+  }
+
+  private void openExportWorkspaceForm() {
```

Comment:
`openExportWorkspaceForm()` could be useful for other tests and so more suitable for `WorkspaceOverview` page object.